### PR TITLE
[CSL-1167] Report error in applyBlocksUnsafe if passed more than epoch

### PR DESCRIPTION
**Pull request by volhovM**

### CODE REVIEW

File: src/Pos/Block/Logic/Internal.hs [hunk 1]
```diff
@@ -111,8 +114,12 @@ type BlockApplyMode ssc m
 applyBlocksUnsafe
     :: forall ssc m . BlockApplyMode ssc m
     => OldestFirst NE (Blund ssc) -> Maybe PollModifier -> m ()
-applyBlocksUnsafe blunds0 pModifier =
-    reportingFatal version $
+applyBlocksUnsafe blunds0 pModifier = reportingFatal version $ do
+    when (length blunds /= length blunds0) $ assertionFailed $
```
Comment (gromakovsky) [inline src/Pos/Block/Logic/Internal.hs]: I don't like how it was done. This `spanSafe` is confusing. This function does something only if `blunds0 == blunds`. So `spanSafe` must return `(blunds0, [])`. So why not just check that all blocks have same epoch as the first one (head)? Also these names complicate understanding.

Comment (volhovM) [inline src/Pos/Block/Logic/Internal.hs]: I can instead check that `b` is null in  `(_,b) = spanSafe ...`. Will it be better?

Comment (gromakovsky) [inline src/Pos/Block/Logic/Internal.hs]: Yes, it's better than current approach and I will approve if you do it this way. But also can you rename `blunds0` to `blunds` and `b` to `blundsOtherEpoch` or something more descriptive?

Comment (volhovM) [inline src/Pos/Block/Logic/Internal.hs]: Done.

**Pull request merged**